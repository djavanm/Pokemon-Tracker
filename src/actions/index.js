export const setPokemon = pokemon => ({
  type: 'SET_POKEMON',
  pokemon
});

export const catchPokemon = pokemon => ({
  type: 'CATCH_POKEMON',
  pokemon
});

export const releasePokemon = id => ({
  type: 'RELEASE_POKEMON',
  id
});
