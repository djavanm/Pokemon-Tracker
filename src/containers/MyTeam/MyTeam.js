import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchPokemon } from '../../thunks/fetchPokemon';
import { Link } from 'react-router-dom';
import MiniCard from '../../components/MiniCard/MiniCard';

export class MyTeam extends Component {

  showPokemon = async (e, newId) => {
    e.preventDefault();
    const { fetchPokemon } = this.props;
    const url = `https://pokeapi.co/api/v2/pokemon/${newId}/`;
    fetchPokemon(url);
  };

  render() {
    const { teamPokemon } = this.props;
    let teamGifs = teamPokemon.map((pokemon, index) => {
      return <MiniCard key={pokemon.id} pokemon={pokemon} showPokemon={this.showPokemon} />
    });
    return (
      <section className='my-team-container'>
      { teamPokemon.length === 0 &&
       <div className='warning-div'>
       <p>Go to myPC to add your pokemon to your team!</p>
       <Link to='/myPC'><button className="team-btn">Catch Pokemon</button></Link>
       <Link to='/myPC'><button className="team-btn">Add Pokemon to Team </button></Link>
      </div> }
      {teamGifs}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  teamPokemon: state.teamPokemon
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({fetchPokemon}, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(MyTeam);
