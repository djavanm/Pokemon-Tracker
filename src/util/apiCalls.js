export const getPokemon = async indentifier => {
  const url = `https://pokeapi.co/api/v2/pokemon/${indentifier}/`;
  const response = await fetch(url);
  if(!response.ok) {
    throw new Error('There was an error getting your pokemon.');
  };
  const pokemon = await response.json();
  return pokemon;
};
