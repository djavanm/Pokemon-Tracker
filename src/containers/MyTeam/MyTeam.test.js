import React, { Component } from 'react';
import { shallow } from 'enzyme';
import { MyTeam, mapStateToProps, mapDispatchToProps } from './MyTeam';
import { getPokemon, getDescription } from '../../util/apiCalls';
import { setPokemon } from '../../actions';
import { Link } from 'react-router-dom';

jest.mock('../../util/apiCalls');

getDescription.mockImplementation(() => {
  return 'Bulbasaur is cool.';
});

getPokemon.mockImplementation(() => {
  return {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
  };
});

describe('MyTeam', () => {
  let wrapper;
  const mockPokemon = {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
    types: ['Grass', 'Poison']
  };

  const mockCaughtPokemon = [mockPokemon];
  const mockTeamPokemon = [mockPokemon];
  const mockCurrentPokemon =  mockPokemon;
  const mockState = {
    currentPokemon: mockPokemon,
    caughtPokemon: mockCaughtPokemon,
    teamPokemon: mockTeamPokemon
  };

  beforeEach(() => {
    wrapper = shallow(<MyTeam
      teamPokemon={mockTeamPokemon}
      />)
  });

  it('it should match the snapshot given the correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      teamPokemon: mockTeamPokemon
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('should get a new pokemon when the show Pokemon method is called', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.instance().showPokemon(mockEvent, 1);
    expect(getPokemon).toHaveBeenCalledWith(1);
    wrapper.instance().showPokemon(mockEvent, 'charizard');
    expect(getPokemon).toHaveBeenCalledWith('charizard');
  });

});
