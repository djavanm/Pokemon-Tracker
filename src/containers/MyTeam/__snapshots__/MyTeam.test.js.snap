// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTeam it should match the snapshot given the correct data 1`] = `
<section
  className="my-team-container"
>
  <MiniCard
    key="1"
    pokemon={
      Object {
        "description": "Bulbasaur is cool.",
        "gen": "Generation 1",
        "gif": "https://projectpokemon.org/images/normal-sprite/bulbasaur.gif",
        "height": 7,
        "id": 1,
        "image": "https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png",
        "name": "Bulbasaur",
        "sprite": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "types": Array [
          "Grass",
          "Poison",
        ],
      }
    }
    showPokemon={[Function]}
  />
</section>
`;
